packages/compass-components/src/hooks/use-confirmation.tsx

```diff
@@ -0,0 +1,147 @@
+    return new Promise<boolean>((resolve) => {
+      const event = new ShowConfirmationEvent(props, resolve);
+      this.dispatchEvent(event);
+    });
+  }
+}
+const globalConfirmation = new GlobalConfirmation();
+export const showConfirmation =
+  globalConfirmation.showConfirmation.bind(globalConfirmation);
+
+type ConfirmationModalAreaProps = ConfirmationProperties & {
+  open: boolean;
+};
+
+export const ConfirmationModalArea: React.FC = ({ children }) => {
+  const hasParentContext = useContext(ConfirmationModalContext).isMounted;
+
+  const [confirmationProps, setConfirmationProps] = useState({
+    open: false,
+  } as ConfirmationModalAreaProps);
+  const callbackRef = useRef<(value: boolean) => void>();
+
+  useEffect(() => {
+    return () => {
+      callbackRef.current?.(false);
+    };
+  }, []);
+
+  // Event listener to use confirmation modal outside of react
+  useEffect(() => {
+    const listener = ({ props, callback }: ShowConfirmationEvent) => {
+      showConfirmation(props, callback);
+    };
+    globalConfirmation.addEventListener(
+      'show-confirmation',
+      listener as EventListener
+    );
+    return () => {
+      globalConfirmation.removeEventListener(
+        'show-confirmation',
+        listener as EventListener
+      );
+    };
+  }, []);
+
+  const showConfirmation = (
+    props: ConfirmationProperties,
+    callback: ConfirmationCallback
+  ) => {
+    if (confirmationProps.open) {
+      throw new Error('Confirmation modal is already open');
+    }
+    setConfirmationProps({ open: true, ...props });
+    callbackRef.current = callback;
+  };
+
+  const handleConfirm = () => {
+    onUserAction(true);
+  };
+
+  const handleCancel = () => {
+    onUserAction(false);
+  };
+
+  const onUserAction = (value: boolean) => {
+    setConfirmationProps((state) => ({ ...state, open: false }));
+    callbackRef.current?.(value);
+    callbackRef.current = undefined;
+  };
+
+  if (hasParentContext) {
+    return <>{children}</>;
+  }
+
+  return (
+    <ConfirmationModalContext.Provider
+      value={{ showConfirmation, isMounted: true }}
+    >
+      {children}
+      <ConfirmationModal
+        data-testid="confirmation-modal"
+        open={confirmationProps.open}
+        title={confirmationProps.title}
+        variant={confirmationProps.variant ?? ConfirmationModalVariant.Default}
+        buttonText={confirmationProps.buttonText ?? 'Confirm'}
+        onConfirm={handleConfirm}
+        onCancel={handleCancel}
+      >
+        {confirmationProps.description}
+      </ConfirmationModal>
+    </ConfirmationModalContext.Provider>
+  );
+};
+
+export const useConfirmationModal = () => {
+  const { showConfirmation: requestConfirmation } = useContext(
+    ConfirmationModalContext
+  );
+  const showConfirmation = (props: ConfirmationProperties) => {
+    return new Promise<boolean>((resolve) =>
+      requestConfirmation(props, resolve)
+    );
+  };
```

Comment:
A bit weird to me that we are re-wrapping this method, why can't we do expected implementation in one place? Makes it really hard to understand what will be the behavior of this method